docker-compose.yml

```
@@ -53,34 +63,35 @@ services:
     links:
     - mariadb:mariadb
     - redis:redis
+    environment: *bb_env
     env_file:
     - .env
     restart: on-failure
     command: django-admin rqworker
     depends_on:
     - mariadb
     - redis
-    links:
-    - mariadb:mariadb
-    - redis:redis
     volumes:
     - .:/app
 
   mariadb:
     image: mariadb
-    env_file:
-    - .env
+    environment:
+      MARIADB_ROOT_PASSWORD: toor
```

In addition, we could specify default inline values for service-dependent environment variables, as well as our service-independent global variables like `DB_PASSWORD`:
```suggestion
      MARIADB_ROOT_PASSWORD: ${DB_PASSWORD:-toor}
```
This suggestion would simplify the understanding for developers who are not familiar with Docker's [environment variables precedence](https://docs.docker.com/compose/envvars-precedence/). Further, this suggestion automatically documents environment variables that could be set in a Docker's `*.env` file if the inlined default values shouldn't be evaluated.